upstairs/src/client.rs

```diff
@@ -896,7 +895,7 @@ impl DownstairsClient {
             // client back into compliance by replaying jobs).
             DsState::Active => EnqueueResult::Send,
             DsState::Connecting {
-                mode: ConnectionMode::Offline,
+                mode: ConnectionMode::Offline | ConnectionMode::New, // RO client checked above
```

Comment:
I think this comment is saying that if we're in this match, it's a read-write config right? Because we'd have executed the first match if it was read-only?

Reply:
Yes exactly.
I wanted to help if someone noticed that `ConnectionMode::New` appeared in two places.

Reply:
There's a behavior change here for the read-write case.

Previously, it would panic if you tried to send a message to a downstairs in `Connecting { mode: New, .. }`.  Now, it will either return `Skip` (if read-only) or `Hold` (if read-write).  I can believe that the read-only behavior change is intended; is the read-write behavior change also intentional?

Reply:
> is the read-write behavior change also intentional?

No!

I put this back in the wrong place, let me fix.

Reply:
Fixed in f212546c